src/Microsoft.Health.Fhir.Api/Features/Headers/FhirHeaders.cs

```diff
@@ -31,7 +33,7 @@ public static FhirResult SetLocationHeader(this FhirResult fhirResult, IUrlResol
 
         public static FhirResult SetETagHeader(this FhirResult fhirResult)
         {
-            var resource = fhirResult.Resource;
+            var resource = fhirResult.Resource.Instance.ToPoco<Resource>();
```

Comment:
Don't need to convert this back to poco. Can just access the versionId through `resource.VersionId`.